feat: Add support for Dart 3.1, remove support for Dart 2.18 (#969)
parse-community/Parse-SDK-Flutter: packages/dart/README.md

## Compatibility

The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to ensure compatibility. To give developers time to upgrade their app to a newer Dart framework, previous Dart framework releases are supported for at least 1 year after the [release date](https://dart.dev/get-dart/archive) of the next higher significant version.

| Version   | Latest Version | End of Support | Compatible |
|-----------|----------------|----------------|------------|
| Dart 3.1  | 3.1.2          | Sep 2024       | ✅ Yes      |
| Dart 3.0  | 3.0.7          | May 2024       | ✅ Yes      |

> | Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |

Reviewer: ```suggestion
| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |
```

> | Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |
> 

Reviewer: ```suggestion
| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |

```

Author: According the removal of the outdated version, I think it is better to remove the Compatible tab

Reviewer: The tab has a different purpose, it's a standard tab in all our repositories.

## Compatibility

The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to ensure compatibility. To give developers time to upgrade their app to a newer Dart framework, previous Dart framework releases are supported for at least 1 year after the [release date](https://dart.dev/get-dart/archive) of the next higher significant version.

| Version   | Latest Version | End of Support | Compatible |
|-----------|----------------|----------------|------------|
| Dart 3.1  | 3.1.2          | Sep 2024       | ✅ Yes      |
| Dart 3.0  | 3.0.7          | May 2024       | ✅ Yes      |
| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |

> | Dart 2.18 | 2.18.5         | Nov 2023       | ❌ No       |

Reviewer: ```suggestion

```

## Compatibility

The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to ensure compatibility. To give developers time to upgrade their app to a newer Dart framework, previous Dart framework releases are supported for at least 1 year after the [release date](https://dart.dev/get-dart/archive) of the next higher significant version.

| Version   | Latest Version | End of Support | Compatible |
|-----------|----------------|----------------|------------|
| Dart 3.1  | 3.1.2          | Sep 2024       | ✅ Yes      |
| Dart 3.0  | 3.0.7          | May 2024       | ✅ Yes      |
| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |

> | Dart 2.18 | 2.18.5         | Nov 2023       | ❌ No       |
> 
> > Parse Dart SDK requires **Dart >=2.19.6**

Reviewer: This extra line is covered by the table above.

```suggestion
| Dart 2.18 | 2.18.5         | Nov 2023       | ❌ No       |

```

## Compatibility

The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to
ensure compatibility. To give developers time to upgrade their app to the newest Dart framework,
previous Dart framework releases are supported for at least 1 year after
their [release date](https://dart.dev/get-dart/archive).

| Version   | Latest Version | End of Support | Compatible |
|-----------|----------------|----------------|------------|
| Dart 3.1  | 3.1.2          | Sep 2024       | ✅ Yes      |
| Dart 3.0  | 3.0.7          | May 2024       | ✅ Yes      |
| Dart 2.19 | 2.19.6         | Mar 2024       | ✅ Yes      |
| Dart 2.18 | 2.18.5         | Nov 2023       | ❌ No       |

> Parse Dart SDK requires **Dart >=2.19.6**

## Getting Started

To install, add the Parse Dart SDK as
a [dependency](https://pub.dev/packages/parse_server_sdk/install) in your `pubspec.yaml` file.

## Documentation

Find the full documentation in the [Parse Dart SDK guide][guide].

## Contributing

We want to make contributing to this project as easy and transparent as possible. Please refer to

> the [Contribution Guidelines](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/CONTRIBUTING.md).

Reviewer: There seems to be some issue with your automated code formatting (if you haven't made these changes manually). There should not be any line breaks, these are added automatically as needed by the markdown renderer.

Author: Yes, it changed, please check again

Reviewer: looks good

> The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to ensure compatibility. To give developers time to upgrade their app to the newest Dart framework, previous Dart framework releases are supported for at least 1 year after their [release date](https://dart.dev/get-dart/archive).

Reviewer: ```suggestion
The Parse Dart SDK is continuously tested with the most recent release of the Dart framework to ensure compatibility. To give developers time to upgrade their app to a newer Dart framework, previous Dart framework releases are supported for at least 1 year after the [release date](https://dart.dev/get-dart/archive) of the next higher significant version.

```

We want to make contributing to this project as easy and transparent as possible. Please refer to the [Contribution Guidelines](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/CONTRIBUTING.md).

> [guide]: https://docs.parseplatform.org/dart/guide/
> 
> [open-collective-link]: https://opencollective.com/parse-server

Reviewer: ```suggestion
[guide]: https://docs.parseplatform.org/dart/guide/
[open-collective-link]: https://opencollective.com/parse-server

```
